Makefile

```diff
@@ -44,61 +37,26 @@ local: build-dirs
 	OUTPUT_DIR=$$(pwd)/_output/bin/$(GOOS)/$(GOARCH) \
 	./hack/build.sh
 
-build: _output/bin/$(GOOS)/$(GOARCH)/$(BIN) ./$(BIN)
-
-_output/bin/$(GOOS)/$(GOARCH)/$(BIN): build-dirs
-	@echo "building: $@"
-	$(MAKE) shell CMD="-c '\
-		GOOS=$(GOOS) \
-		GOARCH=$(GOARCH) \
-		PKG=$(PKG) \
-		BIN=$(BIN) \
-		OUTPUT_DIR=/output/$(GOOS)/$(GOARCH) \
-		./hack/build.sh'"
-
-TTY := $(shell tty -s && echo "-t")
+# test runs unit tests using 'go test' in the local environment.
+test:
+	CGO_ENABLED=0 go test -v -timeout 60s ./...
 
-shell: build-dirs
-	@echo "running docker: $@"
-	@docker run \
-		-e GOFLAGS \
-		-i $(TTY) \
-		--rm \
-		-u $$(id -u):$$(id -g) \
-		-v $$(pwd)/.go/pkg:/go/pkg \
-		-v $$(pwd)/.go/src:/go/src \
-		-v $$(pwd)/.go/std:/go/std \
-		-v $$(pwd):/go/src/$(PKG) \
-		-v $$(pwd)/.go/std/$(GOOS)_$(GOARCH):/usr/local/go/pkg/$(GOOS)_$(GOARCH)_static \
-		-v "$$(pwd)/.go/go-build:/.cache/go-build:delegated" \
-		-e CGO_ENABLED=0 \
-		-w /go/src/$(PKG) \
-		$(BUILD_IMAGE) \
-		go build -installsuffix "static" -i -v -o _output/bin/$(GOOS)/$(GOARCH)/$(BIN) ./$(BIN)
+# ci is a convenience target for CI builds.
+ci: test
 
-build-dirs:
-	@mkdir -p _output/bin/$(GOOS)/$(GOARCH)
-	@mkdir -p .go/src/$(PKG) .go/pkg .go/bin .go/std/$(GOOS)/$(GOARCH) .go/go-build
+# container builds a Docker image containing the binary.
+container:
+	docker build -t $(IMAGE):$(VERSION) .
 
-container: all
-	cp Dockerfile _output/bin/$(GOOS)/$(GOARCH)/Dockerfile
-	docker build -t $(IMAGE):$(VERSION) -f _output/bin/$(GOOS)/$(GOARCH)/Dockerfile _output/bin/$(GOOS)/$(GOARCH)
-	
+# push pushes the Docker image to its repository.
 push:
 	@docker push $(IMAGE):$(VERSION)
 
-container-name:
-	@echo "container: $(IMAGE):$(VERSION)"
-
-all-ci: $(addprefix ci-, $(BIN))
-
-ci-%:
-	$(MAKE) --no-print-directory BIN=$* ci
-
-ci:
-	mkdir -p _output
-	CGO_ENABLED=0 go test -v -timeout 60s ./...
+# build-dirs creates the necessary directories for a build in the local environment.
+build-dirs:
+	@mkdir -p _output/bin/$(GOOS)/$(GOARCH)
```

Comment:
Just curious, why do we put the binary here? Wouldn't it be simpler to place the built binary in the root dir? I'm also not sure that we need multiple dirs for different OS/archs, aren't we always building one OS/arch?

Reply:
yeah, this part is more or less copied over from the main velero makefile, which is copied from some upstream repo. I think it'd be fine to just output to the root dir

Reply:
actually, right now we can't output the binary in the root dir because we already have a subdir there with the same name (`velero-plugin-for-aws/`). So, I can:

- leave this as-is
- output the binary directly under `_output`
- something else

Preferences?

Reply:
I prefer for it not to be at the root. Directly under `_output/` sounds good to me.

Reply:
done